src/Task.ts

```diff
@@ -537,11 +536,10 @@ export class Task {
      * If the global filter exists as part of a nested tag, we keep it untouched.
      */
     public getDescriptionWithoutGlobalFilter() {
-        const { globalFilter } = getSettings();
         let description = this.description;
-        if (globalFilter.length === 0) return description;
+        if (GlobalFilter.isEmpty()) return description;
         // This matches the global filter (after escaping it) only when it's a complete word
-        const globalFilterRegex = RegExp('(^|\\s)' + RegExpTools.escapeRegExp(globalFilter) + '($|\\s)', 'ug');
+        const globalFilterRegex = RegExp('(^|\\s)' + RegExpTools.escapeRegExp(GlobalFilter.get()) + '($|\\s)', 'ug');
```

Comment:
I realised you are probably still working on this, but this is thing thing I was referring to, where I think there should be an explicit method in GlobalFilter to do this work - to remove the `.get()` call.